src/Microsoft.PowerShell.PSReadLine/ReadLine.cs

```diff
@@ -881,30 +882,54 @@ public static void InvokePrompt(ConsoleKeyInfo? key = null, object arg = null)
             _singleton.Render();
         }
 
+        private const string s_PromptCommand = "prompt";
+        private const string s_DefaultPrompt = "PS>";
+
         /// <summary>
         /// Gets the current prompt as possibly defined by the user through the
         /// prompt function, and returns a default prompt if no other is
         /// available. Also handles remote prompts.
         /// </summary>
         public static string GetPrompt()
         {
+            string newPrompt;
             var runspaceIsRemote = _singleton._mockableMethods.RunspaceIsRemote(_singleton._runspace);
-            System.Management.Automation.PowerShell ps;
-            if (!runspaceIsRemote)
+
+            if ((_singleton._runspace.Debugger != null) && _singleton._runspace.Debugger.InBreakpoint)
             {
-                ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                // Run prompt command in debugger API to ensure it is run correctly on the runspace.
+                // This handles remote runspace debugging and nested debugger scenarios.
+                PSDataCollection<PSObject> results = new PSDataCollection<PSObject>();
+                var command = new PSCommand();
+                command.AddCommand(s_PromptCommand);
+                _singleton._runspace.Debugger.ProcessCommand(
+                    command,
+                    results);
+
+                newPrompt = (results.Count == 1) ? (results[0].BaseObject as string) : s_DefaultPrompt;
+                if (newPrompt == null)
+                {
+                    newPrompt = s_DefaultPrompt;
+                }
             }
```

Comment:
This should be moved outside the if/else so it applies to the normal case as well. It should probably also be `string.IsNullOrEmpty(newPrompt)`